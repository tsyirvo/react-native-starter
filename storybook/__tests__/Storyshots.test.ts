import initStoryshots from '@storybook/addon-storyshots';

import render from '$tests/utils';

initStoryshots({
  renderer: render,
});
