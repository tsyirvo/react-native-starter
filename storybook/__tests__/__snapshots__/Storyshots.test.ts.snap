// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Box With a size, color and radiuses 1`] = `
<View
  alignItems="center"
  justifyContent="center"
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    bg="grey"
    borderRadius="medium"
    size={100}
    style={
      Array [
        Object {
          "backgroundColor": "#e5e5e5",
          "borderRadius": 16,
          "height": 100,
          "width": 100,
        },
      ]
    }
  />
</View>
`;

exports[`Storyshots Box With border props and full width 1`] = `
<View
  alignItems="center"
  justifyContent="center"
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    bg="grey"
    borderBottomColor="green"
    borderBottomWidth={4}
    height={50}
    style={
      Array [
        Object {
          "backgroundColor": "#e5e5e5",
          "borderBottomColor": "#2a9d8f",
          "borderBottomWidth": 4,
          "height": 50,
          "width": "100%",
        },
      ]
    }
    width="100%"
  />
</View>
`;

exports[`Storyshots Box With position props and odd size 1`] = `
<View
  alignItems="center"
  justifyContent="center"
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    bg="grey"
    height={50}
    left={20}
    mb={50}
    style={
      Array [
        Object {
          "backgroundColor": "#e5e5e5",
          "height": 50,
          "left": 20,
          "marginBottom": 50,
          "top": 125,
          "width": 100,
        },
      ]
    }
    top={125}
    width={100}
  />
</View>
`;

exports[`Storyshots Button With a text content 1`] = `
<View
  alignItems="center"
  justifyContent="center"
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#e5e5e5",
          "borderRadius": 16,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        },
      ]
    }
    variant="base"
  >
    <Text
      color="black"
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "lineHeight": 24,
          },
        ]
      }
      variant="regular"
    >
      Click here
    </Text>
  </View>
</View>
`;

exports[`Storyshots FallbackLoader With a 3 seconds delay 1`] = `
<View
  alignItems="center"
  justifyContent="center"
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
/>
`;

exports[`Storyshots FallbackLoader With default props 1`] = `
<View
  alignItems="center"
  justifyContent="center"
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
/>
`;

exports[`Storyshots FallbackLoader Without delay 1`] = `
<View
  alignItems="center"
  justifyContent="center"
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
/>
`;

exports[`Storyshots Input Not editable 1`] = `
<View
  alignItems="center"
  alignSelf="center"
  flex={1}
  justifyContent="center"
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
        "width": "80%",
      },
    ]
  }
  width="80%"
>
  <View
    mt="medium"
    style={
      Array [
        Object {
          "marginTop": 16,
          "width": "80%",
        },
      ]
    }
    width="80%"
  >
    <Text
      color="black"
      mb="small"
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "lineHeight": 24,
            "marginBottom": 8,
          },
        ]
      }
      testID="input-label"
      variant="regular"
    >
      Some label
    </Text>
    <TextInput
      allowFontScaling={true}
      editable={false}
      placeholder="Type here"
      placeholderTextColor="#e5e5e5"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "borderBottomColor": "#e5e5e5",
          },
          Object {
            "borderBottomWidth": 1,
            "fontSize": 16,
            "padding": 0,
            "paddingBottom": 5,
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;

exports[`Storyshots Input With a label 1`] = `
<View
  alignItems="center"
  alignSelf="center"
  flex={1}
  justifyContent="center"
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
        "width": "80%",
      },
    ]
  }
  width="80%"
>
  <View
    mt="medium"
    style={
      Array [
        Object {
          "marginTop": 16,
          "width": "80%",
        },
      ]
    }
    width="80%"
  >
    <Text
      color="black"
      mb="small"
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "lineHeight": 24,
            "marginBottom": 8,
          },
        ]
      }
      testID="input-label"
      variant="regular"
    >
      Some label
    </Text>
    <TextInput
      allowFontScaling={true}
      editable={true}
      placeholder="Type here"
      placeholderTextColor="#e5e5e5"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "borderBottomColor": "#e5e5e5",
          },
          Object {
            "borderBottomWidth": 1,
            "fontSize": 16,
            "padding": 0,
            "paddingBottom": 5,
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;

exports[`Storyshots Input With an error 1`] = `
<View
  alignItems="center"
  alignSelf="center"
  flex={1}
  justifyContent="center"
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
        "width": "80%",
      },
    ]
  }
  width="80%"
>
  <View
    mt="medium"
    style={
      Array [
        Object {
          "marginTop": 16,
          "width": "80%",
        },
      ]
    }
    width="80%"
  >
    <Text
      color="black"
      mb="small"
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "lineHeight": 24,
            "marginBottom": 8,
          },
        ]
      }
      testID="input-label"
      variant="regular"
    >
      Some label
    </Text>
    <TextInput
      allowFontScaling={true}
      editable={true}
      error="An error"
      placeholder="Type here"
      placeholderTextColor="#e5e5e5"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "borderBottomColor": "#ef233c",
          },
          Object {
            "borderBottomWidth": 1,
            "fontSize": 16,
            "padding": 0,
            "paddingBottom": 5,
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
    <View
      alignItems="center"
      flexDirection="row"
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
      testID="input-error"
    >
      <Text
        color="red"
        style={
          Array [
            Object {
              "color": "#ef233c",
              "fontSize": 12,
              "lineHeight": 17,
            },
          ]
        }
        variant="small"
      >
        An error
      </Text>
    </View>
  </View>
</View>
`;

exports[`Storyshots Input Without label 1`] = `
<View
  alignItems="center"
  alignSelf="center"
  flex={1}
  justifyContent="center"
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
        "width": "80%",
      },
    ]
  }
  width="80%"
>
  <View
    mt="medium"
    style={
      Array [
        Object {
          "marginTop": 16,
          "width": "80%",
        },
      ]
    }
    width="80%"
  >
    <TextInput
      allowFontScaling={true}
      editable={true}
      placeholderTextColor="#e5e5e5"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "borderBottomColor": "#e5e5e5",
          },
          Object {
            "borderBottomWidth": 1,
            "fontSize": 16,
            "padding": 0,
            "paddingBottom": 5,
          },
        ]
      }
      underlineColorAndroid="transparent"
      value="Some value"
    />
  </View>
</View>
`;

exports[`Storyshots Text With a custom variant 1`] = `
<View
  alignItems="center"
  justifyContent="center"
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    color="black"
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 24,
          "lineHeight": 33,
        },
      ]
    }
    variant="large"
  >
    Large variant
  </Text>
</View>
`;

exports[`Storyshots Text With custom positionning 1`] = `
<View
  alignItems="center"
  justifyContent="center"
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    color="black"
    mt={100}
    paddingY={10}
    pl={20}
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 16,
          "lineHeight": 24,
          "marginTop": 100,
          "paddingBottom": 10,
          "paddingLeft": 20,
          "paddingTop": 10,
        },
      ]
    }
    variant="regular"
  >
    Custom position
  </Text>
</View>
`;

exports[`Storyshots Text With custom styles 1`] = `
<View
  alignItems="center"
  justifyContent="center"
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    color="blue"
    fontWeight={600}
    lineHeight="50px"
    style={
      Array [
        Object {
          "color": "#457b9d",
          "fontSize": 16,
          "fontWeight": "600",
          "lineHeight": 50,
          "textAlign": "center",
        },
      ]
    }
    textAlign="center"
    variant="regular"
  >
    Custom styles
  </Text>
</View>
`;

exports[`Storyshots Text Without any props 1`] = `
<View
  alignItems="center"
  justifyContent="center"
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    color="black"
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 16,
          "lineHeight": 24,
        },
      ]
    }
    variant="regular"
  >
    Default styles
  </Text>
</View>
`;

exports[`Storyshots Theme Colors 1`] = `
<RCTScrollView>
  <View>
    <View
      alignItems="center"
      justifyContent="center"
      mb={150}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "marginBottom": 150,
          },
        ]
      }
    >
      <View
        alignItems="center"
        mb={25}
        style={
          Array [
            Object {
              "alignItems": "center",
              "marginBottom": 25,
            },
          ]
        }
      >
        <Text
          color="black"
          style={
            Array [
              Object {
                "color": "#000",
                "fontSize": 16,
                "lineHeight": 24,
              },
            ]
          }
          variant="regular"
        >
          black
        </Text>
        <View
          bg="#000"
          size={100}
          style={
            Array [
              Object {
                "backgroundColor": "#000",
                "height": 100,
                "width": 100,
              },
            ]
          }
        />
      </View>
      <View
        alignItems="center"
        mb={25}
        style={
          Array [
            Object {
              "alignItems": "center",
              "marginBottom": 25,
            },
          ]
        }
      >
        <Text
          color="black"
          style={
            Array [
              Object {
                "color": "#000",
                "fontSize": 16,
                "lineHeight": 24,
              },
            ]
          }
          variant="regular"
        >
          white
        </Text>
        <View
          bg="#fff"
          size={100}
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
                "height": 100,
                "width": 100,
              },
            ]
          }
        />
      </View>
      <View
        alignItems="center"
        mb={25}
        style={
          Array [
            Object {
              "alignItems": "center",
              "marginBottom": 25,
            },
          ]
        }
      >
        <Text
          color="black"
          style={
            Array [
              Object {
                "color": "#000",
                "fontSize": 16,
                "lineHeight": 24,
              },
            ]
          }
          variant="regular"
        >
          grey
        </Text>
        <View
          bg="#e5e5e5"
          size={100}
          style={
            Array [
              Object {
                "backgroundColor": "#e5e5e5",
                "height": 100,
                "width": 100,
              },
            ]
          }
        />
      </View>
      <View
        alignItems="center"
        mb={25}
        style={
          Array [
            Object {
              "alignItems": "center",
              "marginBottom": 25,
            },
          ]
        }
      >
        <Text
          color="black"
          style={
            Array [
              Object {
                "color": "#000",
                "fontSize": 16,
                "lineHeight": 24,
              },
            ]
          }
          variant="regular"
        >
          blue
        </Text>
        <View
          bg="#457b9d"
          size={100}
          style={
            Array [
              Object {
                "backgroundColor": "#457b9d",
                "height": 100,
                "width": 100,
              },
            ]
          }
        />
      </View>
      <View
        alignItems="center"
        mb={25}
        style={
          Array [
            Object {
              "alignItems": "center",
              "marginBottom": 25,
            },
          ]
        }
      >
        <Text
          color="black"
          style={
            Array [
              Object {
                "color": "#000",
                "fontSize": 16,
                "lineHeight": 24,
              },
            ]
          }
          variant="regular"
        >
          green
        </Text>
        <View
          bg="#2a9d8f"
          size={100}
          style={
            Array [
              Object {
                "backgroundColor": "#2a9d8f",
                "height": 100,
                "width": 100,
              },
            ]
          }
        />
      </View>
      <View
        alignItems="center"
        mb={25}
        style={
          Array [
            Object {
              "alignItems": "center",
              "marginBottom": 25,
            },
          ]
        }
      >
        <Text
          color="black"
          style={
            Array [
              Object {
                "color": "#000",
                "fontSize": 16,
                "lineHeight": 24,
              },
            ]
          }
          variant="regular"
        >
          yellow
        </Text>
        <View
          bg="#e9c46a"
          size={100}
          style={
            Array [
              Object {
                "backgroundColor": "#e9c46a",
                "height": 100,
                "width": 100,
              },
            ]
          }
        />
      </View>
      <View
        alignItems="center"
        mb={25}
        style={
          Array [
            Object {
              "alignItems": "center",
              "marginBottom": 25,
            },
          ]
        }
      >
        <Text
          color="black"
          style={
            Array [
              Object {
                "color": "#000",
                "fontSize": 16,
                "lineHeight": 24,
              },
            ]
          }
          variant="regular"
        >
          red
        </Text>
        <View
          bg="#ef233c"
          size={100}
          style={
            Array [
              Object {
                "backgroundColor": "#ef233c",
                "height": 100,
                "width": 100,
              },
            ]
          }
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Storyshots Theme FontSizes 1`] = `
<View
  alignItems="center"
  justifyContent="center"
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    color="black"
    mb="medium"
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 12,
          "lineHeight": 17,
          "marginBottom": 16,
        },
      ]
    }
    variant="small"
  >
    This is a small text
  </Text>
  <Text
    color="black"
    mb="medium"
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 14,
          "lineHeight": 24,
          "marginBottom": 16,
        },
      ]
    }
    variant="medium"
  >
    This is a medium text
  </Text>
  <Text
    color="black"
    mb="medium"
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 16,
          "lineHeight": 24,
          "marginBottom": 16,
        },
      ]
    }
    variant="regular"
  >
    This is a regular text
  </Text>
  <Text
    color="black"
    mb="medium"
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 24,
          "lineHeight": 33,
          "marginBottom": 16,
        },
      ]
    }
    variant="large"
  >
    This is a large text
  </Text>
  <Text
    color="black"
    mb="medium"
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 32,
          "lineHeight": 45,
          "marginBottom": 16,
        },
      ]
    }
    variant="xLarge"
  >
    This is a xLarge text
  </Text>
</View>
`;

exports[`Storyshots Theme Radiuses 1`] = `
<View
  alignItems="center"
  justifyContent="center"
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    alignItems="center"
    mb={25}
    style={
      Array [
        Object {
          "alignItems": "center",
          "marginBottom": 25,
        },
      ]
    }
  >
    <Text
      color="black"
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "lineHeight": 24,
          },
        ]
      }
      variant="regular"
    >
      small
    </Text>
    <View
      bg="grey"
      borderRadius="small"
      height={100}
      style={
        Array [
          Object {
            "backgroundColor": "#e5e5e5",
            "borderRadius": 4,
            "height": 100,
            "width": 100,
          },
        ]
      }
      width={100}
    />
  </View>
  <View
    alignItems="center"
    mb={25}
    style={
      Array [
        Object {
          "alignItems": "center",
          "marginBottom": 25,
        },
      ]
    }
  >
    <Text
      color="black"
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "lineHeight": 24,
          },
        ]
      }
      variant="regular"
    >
      regular
    </Text>
    <View
      bg="grey"
      borderRadius="regular"
      height={100}
      style={
        Array [
          Object {
            "backgroundColor": "#e5e5e5",
            "borderRadius": 8,
            "height": 100,
            "width": 100,
          },
        ]
      }
      width={100}
    />
  </View>
  <View
    alignItems="center"
    mb={25}
    style={
      Array [
        Object {
          "alignItems": "center",
          "marginBottom": 25,
        },
      ]
    }
  >
    <Text
      color="black"
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "lineHeight": 24,
          },
        ]
      }
      variant="regular"
    >
      medium
    </Text>
    <View
      bg="grey"
      borderRadius="medium"
      height={100}
      style={
        Array [
          Object {
            "backgroundColor": "#e5e5e5",
            "borderRadius": 16,
            "height": 100,
            "width": 100,
          },
        ]
      }
      width={100}
    />
  </View>
</View>
`;

exports[`Storyshots Theme Spaces 1`] = `
<View
  alignItems="center"
  justifyContent="center"
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    color="black"
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 16,
          "lineHeight": 24,
        },
      ]
    }
    variant="regular"
  >
    small - 8px
  </Text>
  <View
    bg="grey"
    height={8}
    mb={8}
    style={
      Array [
        Object {
          "backgroundColor": "#e5e5e5",
          "height": 8,
          "marginBottom": 8,
          "width": "100%",
        },
      ]
    }
    width="100%"
  />
  <Text
    color="black"
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 16,
          "lineHeight": 24,
        },
      ]
    }
    variant="regular"
  >
    medium - 16px
  </Text>
  <View
    bg="grey"
    height={16}
    mb={16}
    style={
      Array [
        Object {
          "backgroundColor": "#e5e5e5",
          "height": 16,
          "marginBottom": 16,
          "width": "100%",
        },
      ]
    }
    width="100%"
  />
  <Text
    color="black"
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 16,
          "lineHeight": 24,
        },
      ]
    }
    variant="regular"
  >
    large - 24px
  </Text>
  <View
    bg="grey"
    height={24}
    mb={24}
    style={
      Array [
        Object {
          "backgroundColor": "#e5e5e5",
          "height": 24,
          "marginBottom": 24,
          "width": "100%",
        },
      ]
    }
    width="100%"
  />
  <Text
    color="black"
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 16,
          "lineHeight": 24,
        },
      ]
    }
    variant="regular"
  >
    xLarge - 32px
  </Text>
  <View
    bg="grey"
    height={32}
    mb={32}
    style={
      Array [
        Object {
          "backgroundColor": "#e5e5e5",
          "height": 32,
          "marginBottom": 32,
          "width": "100%",
        },
      ]
    }
    width="100%"
  />
</View>
`;
